### PR TITLE
Enable completions in object binding patterns

### LINKED ISSUE #2676: No completions in an object binding pattern
```TypeScript
interface I {
    x: number;
    y: string;
}

var foo: I;
var { /*1*/ } = foo;
```

Request completion at marker `1`.

Expected: `x` and `y` are completion items.
Actual: No completions available.

### PR DESCRIPTION
**Pull request by DanielRosenwasser**
Fixes #2676.

objectbindingcompletion

### CODE REVIEW

File: src/services/services.ts [hunk 1]
```diff
@@ -3187,17 +3198,16 @@ namespace ts {
...
             }
 
-            function getContainingObjectLiteralApplicableForCompletion(previousToken: Node): ObjectLiteralExpression {
-                // The locations in an object literal expression that are applicable for completion are property name definition locations.
-
-                if (previousToken) {
-                    let parent = previousToken.parent;
-
-                    switch (previousToken.kind) {
+            function getContainingObjectLiteralOrBindingPatternIfApplicableForCompletion(contextToken: Node): ObjectLiteralExpression | BindingPattern {
```
Comment (mhegazy) [inline src/services/services.ts]: can we make this name a little bit better..

**Comment by mhegazy** [comment]
thanks! Looks good..

Can you add a test for filtering when the property name is specified, eg. `var  { a: b, | } = ..` where `a` should not show again in the completion list.

also a test with an array binding pattern with a nested object pattern `let [{a}] = [{a: ""}];`

**Comment by DanielRosenwasser** [comment]
Good idea on the array binding pattern, though I think I already wrote that other test you described a few times. Can you confirm whether that's the case @mhegazy?

**Comment by mhegazy** [comment]
i did not see any test with renamed properties.

**Comment by DanielRosenwasser** [comment]
@mhegazy ah, sorry, I misunderstood. Yes, I'll make one testing both of those as well as one testing parameters.

**Comment by DanielRosenwasser** [comment]
All good now?

File: src/services/services.ts (hunk 1 above)
Comment (DanielRosenwasser) [inline src/services/services.ts]: That's a...way better name. :smile:

**Comment by mhegazy** [comment]
:+1:

**Comment by mhegazy** [comment]
thanks!

**Pull request merged**